packages/atlas/src/components/_nft/NftWidget/NftWidget.hooks.ts

```diff
@@ -0,0 +1,209 @@
+import { differenceInCalendarDays, differenceInSeconds } from 'date-fns'
+import { useEffect } from 'react'
+
+import { useNft, useNftHistory } from '@/api/hooks'
+import { NftWidgetProps } from '@/components/_nft/NftWidget/NftWidget'
+import { useNftState } from '@/hooks/useNftState'
+import { useMemberAvatar } from '@/providers/assets'
+
+import { NftHistoryEntry } from './NftHistory'
+
+type UseNftWidgetReturn = NftWidgetProps | null
+export const useNftWidget = (videoId?: string): UseNftWidgetReturn => {
+  const { nft, nftStatus, startPolling: startNftPolling, stopPolling: stopNftPolling } = useNft(videoId ?? '')
+  const {
+    isOwner,
+    englishTimerState,
+    canWithdrawBid,
+    needsSettling,
+    auctionPlannedEndDate,
+    userBid,
+    startsAtDate,
+    isUserTopBidder,
+    bidFromPreviousAuction,
+    userBidUnlockDate,
+    saleType,
+    startsAtBlock,
+    canChangeBid,
+    isUserWhitelisted,
+    plannedEndAtBlock,
+    hasTimersLoaded,
+  } = useNftState(nft)
+
+  const owner = nft?.ownerMember
+
+  const { url: ownerAvatarUri } = useMemberAvatar(owner)
+  const { url: topBidderAvatarUri } = useMemberAvatar(nftStatus?.status === 'auction' ? nftStatus.topBidder : undefined)
+
+  const {
+    entries: nftHistory,
+    startPolling: startHistoryPolling,
+    stopPolling: stopHistoryPolling,
+  } = useNftHistoryEntries(videoId || null)
+
+  useEffect(() => {
+    const pollingInterval = 10000
+    startNftPolling(pollingInterval)
+    startHistoryPolling(pollingInterval)
+
+    return () => {
+      stopNftPolling()
+      stopHistoryPolling()
+    }
+  }, [startHistoryPolling, startNftPolling, stopHistoryPolling, stopNftPolling])
```

Comment:
I think the better option would be using `pollInterval`. https://www.apollographql.com/docs/react/data/queries/#polling 
We wouldn't need this effect here.

Reply:
Much better this way, thanks!